aws/resource_aws_dms_replication_task.go

```diff
@@ -328,3 +333,29 @@ func resourceAwsDmsReplicationTaskStateRefreshFunc(
 		return v, *v.ReplicationTasks[0].Status, nil
 	}
 }
+
+func cleanReadOnlyReplicationTaskSettings(settings string) (*string, error) {
+	var settingsData map[string]interface{}
+	if err := json.Unmarshal([]byte(settings), &settingsData); err != nil {
+		return nil, err
+	}
+
+	controlTablesSettings, ok := settingsData["ControlTablesSettings"].(map[string]interface{})
+	if ok {
+		delete(controlTablesSettings, "historyTimeslotInMinutes")
```

Comment:
There is another property `HistoryTimeslotInMinutes ` (with a capital H) that is [the real property](https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Tasks.CustomizingTasks.TaskSettings.ControlTable.html). I don't know why this one exists, but I'm cleaning it out since leaving it in when we set state would cause a difference to show.